addons/bus/static/src/services/bus_monitoring_service.js

```diff
@@ -0,0 +1,110 @@
+import { WORKER_STATE } from "@bus/workers/websocket_worker";
+import { reactive } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+import { registry } from "@web/core/registry";
+
+const { DateTime } = luxon;
+
+export const CONNECTION_STATE = Object.freeze({
+    LOST: "LOST",
+    STABLE: "STABLE",
+    UNSTABLE: "UNSTABLE",
+});
+// 2 minutes -- Minimum time a connection must be kept to be considered stable.
+export const STABLE_CONNECTION_TRESHOLD = 120_000;
+
+/**
+ * Monitor the communication bus health. Track connection stability and detect
+ * lost connections. The bus connection is considered:
+ *
+ * - `LOST` if no connection could be established after a reconnect attempt.
+ * - `UNSTABLE` if the last connections was shorter than `STABLE_CONNECTION_TRESHOLD`.
```

Comment:
is longer than `STABLE_CONNECTION_THRESHOLD` I assume?

Reply:
no, it's unstable if we reconnect but the last connection was too short. scenario:
- connect, stays connected a few second
- disconnect
- connect immediately, stays connected a few seconds
- disconnect
and so on.. we are connected, but the connection is clearly unstable.